libdnf5/base/goal.cpp

```diff
@@ -633,11 +663,49 @@ GoalProblem Goal::Impl::add_transaction_replay_specs_to_goal(base::Transaction &
     auto replay_location = replay_path.remove_filename();
     auto replay = transaction::parse_transaction_replay(replay_file.read());
 
+    return add_replay_to_goal(transaction, replay, settings, replay_location);
+}
+
+static std::set<std::string> query_to_vec_of_nevra_str(const libdnf5::rpm::PackageQuery & query) {
+    std::set<std::string> query_set = {};
+    std::transform(query.begin(), query.end(), std::inserter(query_set, query_set.begin()), [](const auto & pkg) {
+        return pkg.to_string();
+    });
+
+    return query_set;
+}
+
+GoalProblem Goal::Impl::add_replay_to_goal(
+    base::Transaction & transaction,
+    const transaction::TransactionReplay & replay,
+    GoalJobSettings settings,
+    std::filesystem::path replay_location) {
+    auto ret = GoalProblem::NO_PROBLEM;
     bool skip_unavailable = settings.resolve_skip_unavailable(base->get_config());
 
+    std::unordered_set<std::string> rpm_nevra_cache;
+
     for (const auto & package_replay : replay.packages) {
+        rpm_nevra_cache.insert(package_replay.nevra);
         libdnf5::GoalJobSettings settings_per_package = settings;
         settings_per_package.set_clean_requirements_on_remove(libdnf5::GoalSetting::SET_FALSE);
+
+        std::optional<libdnf5::rpm::Package> local_pkg;
+        if (!package_replay.package_path.empty()) {
+            local_pkg =
+                base->get_repo_sack()->add_stored_transaction_package(replay_location / package_replay.package_path);
```

Comment:
This path merge is not clear why one path is extended by a prefix in `replay_location`. May I ask you for a comment?

Reply:
Added.